### PR TITLE
refactor: new Buffer() API refactor

### PR DESCRIPTION
**Pull request by mansonchor**
benchmark add more node version

##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: benchmark/format_buffer_vs_string.js [hunk 1]
```diff
@@ -36,12 +36,12 @@ function bufferFormat(level, args, meta, options) {
...
   }
 
-  if (!output) return new Buffer('');
+  if (!output) return new Buffer.from('');
 
   output += options.eol;
 
   // convert string to buffer when encoding is not utf8
-  return options.encoding === 'utf8' ? new Buffer(output) : iconv.encode(output, options.encoding);
+  return options.encoding === 'utf8' ? new Buffer.from(output) : iconv.encode(output, options.encoding);
```
Comment (killagu) [inline benchmark/format_buffer_vs_string.js:44]: new 没删干净

File: benchmark/format_buffer_vs_string.js [hunk 2]
```diff
@@ -36,12 +36,12 @@ function bufferFormat(level, args, meta, options) {
     output = message;
   }
 
-  if (!output) return new Buffer('');
+  if (!output) return new Buffer.from('');
```
Comment (killagu) [inline benchmark/format_buffer_vs_string.js:39]: 这个能用一个常量代替吗？

File: lib/utils.js [hunk 3]
```diff
@@ -102,7 +103,7 @@ module.exports = {
       output = message;
     }
 
-    if (!output) return new Buffer('');
+    if (!output) return new Buffer.from('');
```
Comment (killagu) [inline lib/utils.js:105]: new 没删干净

File: test/lib/formatter.test.js [hunk 4]
```diff
@@ -133,8 +133,8 @@ describe('test/lib/formatter.test.js', () => {
       message: 'should be an integer',
     }];
     err.content = '123\n123';
-    err.buf = new Buffer(1000).fill(0);
-    err.shortBuf = new Buffer(30).fill(101);
+    err.buf = new Buffer.alloc(1000).fill(0);
+    err.shortBuf = new Buffer.alloc(30).fill(101);
```
Comment (killagu) [inline test/lib/formatter.test.js:137]: new 没删

File: benchmark/format_buffer_vs_string.js (hunk 2 above)
Comment (mansonchor) [reply benchmark/format_buffer_vs_string.js]: 意义不是特别大

**Review by killagu — APPROVED** [review]
LGTM

**Pull request merged**